integration/test_ml.py

```diff
@@ -0,0 +1,389 @@
+# Copyright 2020 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Integration tests for firebase_admin.ml module."""
+import re
+import pytest
+
+
+from firebase_admin import ml
+from firebase_admin import exceptions
+from tests import testutils
+
+
+# pylint: disable=import-error,no-name-in-module
+try:
+    import tensorflow as tf
+    import os      # This is only needed for the tensorflow testing
+    import shutil  # This is only needed for the tensorflow testing
+    _TF_ENABLED = True
+except ImportError:
+    _TF_ENABLED = False
+
+
+@pytest.fixture
+def name_only_model():
+    model = ml.Model(display_name="TestModel123")
+    yield model
+
+
+@pytest.fixture
+def name_and_tags_model():
```

Comment:
Most of these fixtures can be made a bit better by organizing them as follows:

```
@pytest.fixture
def name_and_tags_model():
  name = _random_name()
  args = ml.Model(display_name=name, tags=['tag1', 'tag2'])
  model = ml.create_model(args)
  yield model
  ml.delete_model(model.model_id)
```

This simplifies the call-site and also ensures proper clean up. You also don't need to check `_if_exists` which requires an rpc.

Reply:
done then modified further based on conversation.